### PR TITLE
Bucketlist for Peer Review

### PR DESCRIPTION
**Pull request by kjagoo**

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def create_app(configuration):
+    app = Flask(__name__)
+    app.config.from_object(app_config[configuration])
+    db.init_app(app)
+
+    return app
+
+app = create_app("development")
```
Comment (karanjaE) [inline app/__init__.py:16]: Why pass environment manually? Use environment variables then call it using `os.environ`

File: testmanage.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+from flask_script import Manager
+from flask_migrate import Migrate, MigrateCommand
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+
+from datetime import datetime
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql+psycopg2://joshuakagenyi:#@joshua2016@localhost:5432/testbucketlist'
+db = SQLAlchemy(app)
```
Comment (Stephen-Njoroge) [inline testmanage.py:15]: set this in an environment variable. the `app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql+psycopg2://joshuakagenyi:#@joshua2016@localhost:5432/testbucketlist'`

File: manage.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from flask_script import Manager
+from flask_migrate import Migrate, MigrateCommand
+from app import app, db, app_config
+from datetime import datetime
+
+app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql+psycopg2://joshuakagenyi:#@joshua2016@localhost:5432/bucketlist'
+
```
Comment (Stephen-Njoroge) [inline manage.py:9]: this URI would be better as an environment variable next time.

File: app/smain.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+auth = HTTPBasicAuth()
+
+
+@auth.error_handler
+def unauthorized(message=None):
+    """Returns an error message."""
+    if not message:
+        message = "Error: Restricted unauthorised access."
+    return jsonify({"message": message}), 403
+
```
Comment (Stephen-Njoroge) [inline app/smain.py:18]: Unauthorized status code is 401, 403 is forbidden access.

File: tests/test_auth.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        output = json.loads(response.data.decode('utf-8'))
+        self.assertIn("Error: Invalid username and/or password.",
+                      output["message"])
+
+        self.user = {"username": "testuser", "password": "invalid"}
+        response = self.app.post("/auth/login/", data=self.user)
+        self.assertEqual(response.status_code, 403)
+
+        output = json.loads(response.data.decode('utf-8'))
+        self.assertIn("Error: Invalid username and/or password.", output["message"])
```
Comment (Stephen-Njoroge) [inline tests/test_auth.py:42]: Overall, it is nice you decided to go the long way and utilize Postgres database.

**Pull request closed without merging**